python/mxnet/gluon/nn/conv_layers.py

```diff
@@ -1007,3 +1008,32 @@ def __init__(self, layout='NCDHW', **kwargs):
         assert layout == 'NCDHW', "Only supports NCDHW layout for now"
         super(GlobalAvgPool3D, self).__init__(
             (1, 1, 1), None, 0, True, True, 'avg', **kwargs)
+
+
+class ReflectionPad2D(HybridBlock):
+    """Pads the input tensor using the reflection of the input boundary.
+
+    Parameters
+    ----------
+        `padding` is a tuple of integer padding widths for each axis of the format
+``(before_1, after_1, ... , before_N, after_N)``. The `padding` should be of length ``2*N``
+where ``N`` is the number of dimensions of the array.
+
+    Shape:
+        - Input: :math:`(N, C, H_{in}, W_{in})`
+        - Output: :math:`(N, C, H_{out}, W_{out})` where
+          :math:`H_{out} = H_{in} + paddingTop + paddingBottom`
+          :math:`W_{out} = W_{in} + paddingLeft + paddingRight`
+
+    Examples
+    --------
+    >>> m = nn.ReflectionPad2D(3)
+    >>> input = mx.nd.random.normal(shape=(16, 3, 224, 224))
+    >>> output = m(input)
+    """
+    def __init__(self, padding=0, **kwargs):
+        super(ReflectionPad2D, self).__init__(**kwargs)
+        self._padding = padding
+
+    def hybrid_forward(self, F, x):
+        return F.pad(x, mode='reflect', padding=self._padding)
```

Comment:
This isn't tested yet. Also, it looks like a very straightforward one-liner. Should we just ask users to use hybrid lambda for this?

Reply:
This simple API makes user easier to use. The Reflectance padding is popularly used in low level vision, such as GANs, super resolution and style transfer. I think it make sense to provide an API for it. What's your thought?

http://pytorch.org/docs/0.3.0/nn.html?highlight=padding#reflectionpad2d

Reply:
I think we should have this.

Reply:
OK